Detectors/TPC/simulation/src/Detector.cxx

```diff
@@ -62,47 +60,26 @@ using std::ifstream;
 using namespace o2::TPC;
 
 
-Detector::Detector()
-  : o2::Base::Detector("TPC", kTRUE),
-    mSimulationType(SimulationType::Other),
-    mPointCollection(new TClonesArray("o2::TPC::Point")),
-    mHitGroupCollection(new TClonesArray("o2::TPC::LinkableHitGroup")),
-    mGeoFileName(),
-    mEventNr(0)
-{
-  for(int i=0;i<Sector::MAXSECTOR;++i){
-    mHitsPerSectorCollection[i]=new TClonesArray("o2::TPC::LinkableHitGroup");
-    mHitsPerSectorCollection[i]->BypassStreamer(true);
-  }
-}
-
 Detector::Detector(Bool_t active)
   : o2::Base::Detector("TPC", active),
     mSimulationType(SimulationType::Other),
-    mPointCollection(new TClonesArray("o2::TPC::Point")),
-    mHitGroupCollection(new TClonesArray("o2::TPC::LinkableHitGroup")),
     mGeoFileName(),
     mEventNr(0)
 {
   for(int i=0;i<Sector::MAXSECTOR;++i){
-    mHitsPerSectorCollection[i]=new TClonesArray("o2::TPC::LinkableHitGroup");
-    mHitsPerSectorCollection[i]->BypassStreamer(true);
+    mHitsPerSectorCollection[i]=new std::vector<o2::TPC::LinkableHitGroup>;
```

Comment:
If a vector is created in any case for each sector, why not making the mHitsPerSectorCollection on the stack in the first place to avoid calling new. Or use `std::array<std::vector<o2::TPC::LinkableHitGroup>, MAXSECTOR>` in the header?

Reply:
I did not manage to achieve ROOT IO with non-pointer members (yet). So this has to stay.